authorizer.py

```diff
@@ -81,19 +81,21 @@ def HasStudentAccess(self):
   def _VerifyServingSession(self, institution, session):
     serving_session = models.ServingSession.FetchEntity(institution)
     logging.info("currently serving session = %s" % serving_session)
-    if serving_session.session_name != session:
-      logging.error("serving session doesn't match")
-      return False
-    if (serving_session.login_type == "schedule" and
-        self.handler.request.path == "/spots_available"):
+    if serving_session.session_name == session:
       return True
-    if (serving_session.login_type == "preregistration" and
-        self.handler.request.path == "/catalog_print"):
+    logging.error("serving session doesn't match")
+    return False
+
+  def HasPageAccess(self, institution, session, current_page):
```

Comment:
I think this should be removed too. I think it's more clear in the code if each page gets to choose for itself when it can and cannot be displayed. So the code in the page might do the following: 
- check if the user has student access. If not then redirect.
- Then check that the serving level match what we expect in order to display the page. If not then redirect.